contentcloud-spring-boot-autoconfigure/src/test/java/eu/xenit/contentcloud/spring/autoconfigure/s3/S3RegionAutoConfigurationTest.java

```diff
@@ -0,0 +1,118 @@
+package eu.xenit.contentcloud.spring.autoconfigure.s3;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.parallel.Resources.SYSTEM_PROPERTIES;
+
+import java.util.Arrays;
+import java.util.Properties;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.ResourceLock;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+class S3RegionAutoConfigurationTest {
+    private Properties systemPropertiesBackup;
+
+    @BeforeEach
+    void backupSystemProperties() {
+        systemPropertiesBackup = new Properties();
+        systemPropertiesBackup.putAll(System.getProperties());
+    }
+
+    @AfterEach
+    void restoreSystemProperties() {
+        System.setProperties(systemPropertiesBackup);
+    }
+
+    @SneakyThrows
+    private static SdkClientConfiguration extractConfiguration(S3Client s3Client) {
+        Class<? extends S3Client> s3clazz = s3Client.getClass();
+        var sdkClientConfigField = Arrays.stream(s3clazz.getDeclaredFields())
+                .filter(field -> field.getType() == SdkClientConfiguration.class)
+                .findAny()
+                .get();
+
+        sdkClientConfigField.setAccessible(true);
+        return (SdkClientConfiguration) sdkClientConfigField.get(s3Client);
+    }
+
+    @Test
+    @ResourceLock(value = SYSTEM_PROPERTIES)
+    void setsRegionWhenConfiguredExplicitly() {
+        System.setProperty("spring.content.s3.region", "my-region");
+
+        AnnotationConfigApplicationContext applicationContext = new AnnotationConfigApplicationContext();
```

Comment:
FYI you could look into `ApplicationContextRunner` to test auto-configuration - that avoids that hassle with system-properties.

Something along the lines of:
```
new ApplicationContextRunner()
    .withConfiguration(AutoConfigurations.of(S3RegionAutoConfiguration.class))
    .withPropertyValues("spring.content.s3.region=my-region")
    .withUserConfiguration(TestConfig.class)
    .run(context -> {
        S3Client client = context.getBean(S3Client.class);
        SdkClientConfiguration configuration = extractConfiguration(client);
        assertEquals(Region.of("my-region"), configuration.option(AwsClientOption.AWS_REGION));
});
```

References:
* https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.developing-auto-configuration.testing
* https://www.baeldung.com/spring-boot-context-runner
* https://github.com/xenit-eu/thunx/pull/6/files#diff-72f43498992f08ffdb686bac323f0f964b31182a9ae330e8a8a543c463539a29

Reply:
Thanks for the references. I actually took a look at spring-content to see how to test autoconfiguration, but this is a lot better.